django_airavata/apps/api/views.py

```diff
@@ -970,6 +970,16 @@ def get(self, request, format=None):
             data_product, context={'request': request})
         return Response(serializer.data)
 
+    def put(self, request, format=None):
+        data_product_uri = request.query_params['product-uri']
+        data_product = request.airavata_client.getDataProduct(
+            request.authz_token, data_product_uri)
+
+        return UserStoragePathView().put(
+            request=request,
+            path=data_product.replicaLocations[0].filePath.replace("file://localhost:/tmp/default-admin/", "")
```

Comment:
`file://localhost:/tmp/default-admin/` only works based on your local setup, it doesn't generally work. You need a more reliable way to compute the user relative path to the file.

`user_storage.get_rel_path` can return a relative path for a full path. https://github.com/apache/airavata-django-portal-sdk/blob/master/airavata_django_portal_sdk/user_storage.py#L499

However, we need to also implement this for the remote API mode of user_storage, but that can be via a separate issue/PR.

Reply:
Actually user_storage.get_rel_path didn't do that. Instead, I moved this to the serializer and improved the regex